scipy/sparse/_base.py

```diff
@@ -867,29 +871,36 @@ def _getcol(self, j):
         # Subclasses should override this method for efficiency.
         # Post-multiply by a (n x 1) column vector 'a' containing all zeros
         # except for a_j = 1
-        n = self.shape[1]
+        N = self.shape[-1]
         if j < 0:
-            j += n
-        if j < 0 or j >= n:
+            j += N
+        if j < 0 or j >= N:
             raise IndexError("index out of bounds")
         col_selector = self._csc_container(([1], [[j], [0]]),
-                                           shape=(n, 1), dtype=self.dtype)
+                                           shape=(N, 1), dtype=self.dtype)
+        if self.ndim == 1:
+            # only here to force col to be 2d ("as an (mx1) sparse array")
+            # should this be 1d? -- hard to know with name "getcol"
+            return (self @ col_selector).reshape(1, 1)
         return self @ col_selector
 
     def _getrow(self, i):
         """Returns a copy of row i of the array, as a (1 x n) sparse
         array (row vector).
         """
+        if self.ndim == 1:
+            # force col to be 2d ("as an (1xn) sparse array")
+            return self.reshape(1, self.shape[-1])
```

Comment:
I think it makes sense to throw an error here, as `getrow()` isn't really meaningful for a 1d container.